### PR TITLE
Introduce tests for new e-prop models

### PR DESCRIPTION
**Pull request by JesusEV**
This PR Introduces Tests for new e-prop Models

#### Discussion Topics

- **Population Split for Recurrent Neurons:** Is it a good idea to divide the recurrent neurons into three distinct populations: `eprop_iaf`, `eprop_iaf_adapt`, and `eprop_iaf_psc_delta`?

#### To-Do List

- [x]  Update the platform, versions, and branch used for generating the tests.

### CODE REVIEW

**Comment by akorgor** [comment]
Very nice - thanks a lot!  Maybe as two little improvements to reduce code and enlarge the coverage of this test, we could
- [x] use `@pytest.mark.parametrize` to loop over the three neuron models
- [x] use `@pytest.mark.parametrize` to loop over the two optimizers
- [x] maybe test for a batch size of 2 instead of 1 or loop over that parameter as well

Note this functionality to define parameter sets:

```python 
@pytest.mark.parametrize("neuron_model,optimizer", [
    ("eprop_iaf", "Adam"), 
    ("eprop_iaf_psc_delta, "GD"), 
    ("eprop_iaf_adapt", "GD"), 
])
def test_function(neuron_model, optimizer):
```

File: testsuite/pytests/test_eprop_plasticity.py [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+    # Verify results
+
+    if neuron_model == "eprop_iaf":
+        loss_NEST_reference = np.array(
+            [
+                115.18883444998838,
+                102.72225734161191,
+                91.09860416985143,
+                81.78596643400449,
+                67.49056117141924,
```
Comment (akorgor) [inline testsuite/pytests/test_eprop_plasticity.py]: For me, the test was successful only when using this last loss. Could you please test this? If this depends on the machine, we need to adjust the tolerance.
```suggestion
                67.02284542820075,
```

Comment (JesusEV) [reply testsuite/pytests/test_eprop_plasticity.py]: Do any of the other 2 cases fail?

Comment (JesusEV) [reply testsuite/pytests/test_eprop_plasticity.py]: Does any of the other 2 cases fail?

Comment (akorgor) [reply testsuite/pytests/test_eprop_plasticity.py]: No, only that one.

Comment (akorgor) [reply testsuite/pytests/test_eprop_plasticity.py]: Fixed in [27da18c](https://github.com/JesusEV/nest-simulator/pull/24/commits/27da18c31d6f736ccce17c61b4afd158aa38d20c).

**Review by akorgor — APPROVED** [review]

**Pull request merged**